dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkAspect.java

```diff
@@ -155,12 +149,38 @@ private void commitTransaction(UnitOfWork unitOfWork, Session session) {
         }
     }
 
-    protected Session getSession() {
-        return requireNonNull(session);
+    private static Optional<UnitOfWorkContext> getContext() {
+        return Optional.ofNullable(CONTEXT.get());
+    }
+
+    private static void setContext(UnitOfWork unitOfWork, Session session) {
+        ManagedSessionContext.bind(session);
+        CONTEXT.set(new UnitOfWorkContext(unitOfWork, session.getSessionFactory()));
     }
 
-    protected SessionFactory getSessionFactory() {
-        return requireNonNull(sessionFactory);
+    private static void clearContext() {
+        ManagedSessionContext.unbind(getSessionFactory());
+        CONTEXT.remove();
     }
 
+    @Nullable
+    public static SessionFactory getSessionFactory() {
```

Comment:
It is kind of a drag to force clients to handle the nullable - curious if anyone thinks it would be better to throw an exception here if any of these properties are fetched before they are initialized.